src/components/comment/commentContext.js

```diff
@@ -0,0 +1,133 @@
+import React, { createContext, useContext, useState } from 'react';
+import _ from 'lodash';
+
+import * as fetchUtil from '../../util/fetch';
+
+/** @type {React.Context<{commentList:Comment[],commentError:Error,getData:()=>void}} */
+const commentContext = createContext();
+
+/** This is a react component which you wrap your entire application
+ * to provide a "context", meaning: data you can access anywhere in the app.
+ *
+ * @param {object} props
+ * @param {React.ReactChild} props.children
+ * @param {()=>Promise<import('axios').AxiosResponse<Comment[]>>} props.fetchCommentList
+ */
+export const CommentProvider = ({ children }) => {
+  /** @type {Comment[]} */
+  const defaultState = [];
+  const [commentError, setCommentError] = useState();
+  const [lastRefreshCommentList, setLastRefreshCommentList] = useState();
+  const [commentList, setCommentList] = useState(defaultState);
+  const [isLoadingCommentList, setIsLoadingCommentList] = useState(false);
+  const [isLoadingCommentOnPost, setIsLoadingCommentOnPost] = useState(false);
+
+  const refreshCommentList = () => {
+    if (!isLoadingCommentList) {
+      setIsLoadingCommentList(true);
+      // Fetch list of comments
+      fetchUtil.post
+        .fetchCommentList()
+        .then(res => {
+          setCommentList(_.keyBy(res?.data, '_id'));
+        })
+        .catch(err => setCommentError(err))
+        .finally(() => {
+          setLastRefreshCommentList(Date.now);
+          setIsLoadingCommentList(false);
+        });
+    }
+  };
+
+  const commentOnPost = (postId, commentData) =>
+    new Promise(async (resolve, reject) => {
```

Comment:
if you're returning a promise, it's async by definition. You don't need async/await inside a promise.

Reply:
Fixed